src/main/scala/io/qbeast/spark/internal/sources/v2/QbeastBatch.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.internal.sources.v2
+
+import org.apache.spark.sql.connector.read.{Batch, InputPartition, PartitionReaderFactory}
+import org.apache.spark.sql.execution.datasources.FileIndex
+import org.apache.spark.sql.types.StructType
+
+/**
+ * Implementation of org.apache.spark.sql.connector.read.Batch.
+ */
+class QbeastBatch private[v2] (
```

Comment:
Do we need a particular implementation of `Batch`? Could we delegate some of the operations to Parquet built-in classes?

Just to avoid re-implementing a lot of the features, and to be compatible with other possible optimisations for Parquet files.

Reply:
I know we can do it, but I am not sure if it's the best option for the Qbeast reading. 

I am thinking about the special case of Sampling and Limit: since we know what are the min and max weight values of a file, we could read just a part of it.  I don't know if Parquet classes had parameters to infer how many records should be read or if we need to re-implement some of the methods. We can investigate. 

What do you think @cugni ?